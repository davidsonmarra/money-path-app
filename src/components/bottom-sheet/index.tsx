import React from 'react';
import { View, DimensionValue } from 'react-native';
import Modal from 'react-native-modal';
import Text, { TextType } from 'src/components/text';
import useStyles from 'src/components/bottom-sheet/styles';

export interface Props {
  isVisible: boolean;
  onClose: () => void;
  title: string;
  children: React.ReactNode;
  maxHeight?: DimensionValue;
  testID?: string;
}

const BottomSheet = ({
  isVisible,
  onClose,
  title,
  children,
  maxHeight = '80%',
  testID,
}: Props) => {
  const styles = useStyles({ maxHeight });

  return (
    <Modal
      isVisible={isVisible}
      onBackdropPress={onClose}
      onBackButtonPress={onClose}
      style={styles.modal}
      animationIn="slideInUp"
      animationOut="slideOutDown"
      animationInTiming={300}
      animationOutTiming={300}
      backdropTransitionInTiming={300}
      backdropTransitionOutTiming={300}
      testID={testID}
    >
      <View style={styles.bottomSheet}>
        <View style={styles.header}>
          <Text type={TextType.headingMedium} style={styles.headerTitle}>
            {title}
          </Text>
        </View>
        <View style={styles.content}>{children}</View>
      </View>
    </Modal>
  );
};

export default BottomSheet;
