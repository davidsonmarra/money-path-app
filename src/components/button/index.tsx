import React, {ReactNode} from 'react';
import {
  ActivityIndicator,
  TouchableOpacity,
  TouchableOpacityProps,
  View,
} from 'react-native';
import {colors} from 'src/configs/theme';
import Text, {TextType} from 'src/components/text';
import getStyles from 'src/components/button/styles';

export enum ButtonType {
  primary = 'primary',
  secondary = 'secondary',
}

export interface Props extends TouchableOpacityProps {
  text: string;
  leftIcon?: ReactNode;
  type?: ButtonType;
  isLoading?: boolean;
  isDisabled?: boolean;
}

const buttonProps = (isDisabled: boolean) => ({
  [ButtonType.primary]: {
    buttonStyle: getStyles({isDisabled}).primary,
    textType: TextType.buttonPrimary,
  },
  [ButtonType.secondary]: {
    buttonStyle: getStyles({isDisabled}).secondary,
    textType: TextType.buttonSecondary,
  },
});

const Button = ({
  text,
  type = ButtonType.primary,
  isLoading = false,
  isDisabled = false,
  leftIcon,
  style,
  ...rest
}: Props) => {
  return (
    <TouchableOpacity
      disabled={isDisabled || isLoading}
      style={[
        getStyles({isDisabled}).container,
        buttonProps(isDisabled)[type].buttonStyle,
        style,
      ]}
      {...rest}>
      <View>{leftIcon}</View>
      {isLoading ? (
        <ActivityIndicator size={38} color={colors.buttonText} />
      ) : (
        <Text type={buttonProps(isDisabled)[type].textType}>{text}</Text>
      )}
      <View />
    </TouchableOpacity>
  );
};

export default Button;
