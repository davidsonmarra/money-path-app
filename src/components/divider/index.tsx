import React from 'react';
import {View} from 'react-native';
import useStyles from './styles';

export enum DividerType {
  horizontal = 'horizontal',
  vertical = 'vertical',
}

export enum DividerSize {
  small = 'small',
  medium = 'medium',
  large = 'large',
}

export interface Props {
  type?: DividerType;
  size?: DividerSize;
}

const getSize = () => ({
  [DividerSize.small]: 1,
  [DividerSize.medium]: 2.5,
  [DividerSize.large]: 5,
});

const Divider = ({
  type = DividerType.horizontal,
  size = DividerSize.medium,
}: Props) => {
  const dividerSize = getSize()[size];
  const styles = useStyles({size: dividerSize});

  const getDividerStyle = () => ({
    [DividerType.horizontal]: styles.horizontal,
    [DividerType.vertical]: styles.vertical,
  });

  return <View style={getDividerStyle()[type]} />;
};

export default Divider;
