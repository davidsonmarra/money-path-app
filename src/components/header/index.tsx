import React from 'react';
import {TouchableOpacity, View} from 'react-native';
import {useSafeAreaInsets} from 'react-native-safe-area-context';
import Text, {TextType} from 'src/components/text';
import {BackIcon} from 'src/assets/icons';
import useStyles from 'src/components/header/styles';

interface renderIconProps {
  color?: string;
}

export interface Props {
  text?: string;
  renderLeftIcon?: (props?: renderIconProps) => React.ReactNode;
  onLeftIconPress?: () => void;
}

const Header = ({
  text = '',
  renderLeftIcon = props => <BackIcon {...props} />,
  onLeftIconPress,
}: Props) => {
  const insets = useSafeAreaInsets();
  const style = useStyles({insetTop: insets.top});

  return (
    <View style={style.container}>
      <TouchableOpacity onPress={onLeftIconPress} testID="btn-left-icon">
        {renderLeftIcon({
          color: style.textColor.color,
        })}
      </TouchableOpacity>
      <Text type={TextType.headingXSmall} style={style.textColor}>
        {text}
      </Text>
      <View />
    </View>
  );
};

export default Header;
