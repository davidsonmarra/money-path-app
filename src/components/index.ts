export { default as Button, ButtonType } from 'src/components/button';
export {
  default as Divider,
  DividerSize,
  DividerType,
} from 'src/components/divider';
export { default as Feedback, FeedbackType } from 'src/components/feedback';
export { default as Header } from 'src/components/header';
export { default as Input } from 'src/components/input';
export { default as TextArea } from 'src/components/text-area';
export { default as ListItem } from 'src/components/list-item';
export { default as ProfileHeader } from 'src/components/profile-header';
export { default as Spacer } from 'src/components/spacer';
export { default as Text, TextType } from 'src/components/text';
export { default as IconRound } from 'src/components/icon-round';
export { default as StorybookView } from 'src/components/storybook-view';
export { default as SelectionGroup } from 'src/components/selection-group';
