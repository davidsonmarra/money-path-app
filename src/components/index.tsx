export {default as Button, ButtonType} from 'src/components/button';
export {default as Text, TextType} from 'src/components/text';
export {default as StorybookView} from 'src/components/storybook-view';
