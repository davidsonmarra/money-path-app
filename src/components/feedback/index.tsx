import React from 'react';
import {View} from 'react-native';
import {IconType} from 'src/assets/icons/types';
import renderIcon from 'src/assets/icons/utils';
import {Button, ButtonType, Text, TextType} from 'src/components';
import useStyles from 'src/components/feedback/styles';
import {useTheme} from 'src/hooks/useTheme';

export enum FeedbackType {
  success = 'success',
  error = 'error',
  warning = 'warning',
}

export interface Props {
  type?: FeedbackType;
  title: string;
  description?: string;
  buttonText: string;
  onPress: () => void;
}

const Feedback = ({
  type = FeedbackType.success,
  title,
  description,
  buttonText,
  onPress,
}: Props) => {
  const {colors} = useTheme().theme;
  const styles = useStyles({});

  const renderIconFeedback = (type: FeedbackType) =>
    ({
      [FeedbackType.success]: (
        <View style={[styles.feedbackIconWrapper, styles.successWrapper]}>
          {renderIcon(IconType.check)({size: 100, color: colors.success})}
        </View>
      ),
      [FeedbackType.error]: (
        <View style={[styles.feedbackIconWrapper, styles.errorWrapper]}>
          {renderIcon(IconType.error)({size: 100, color: colors.error})}
        </View>
      ),
      [FeedbackType.warning]: (
        <View style={[styles.feedbackIconWrapper, styles.warningWrapper]}>
          {renderIcon(IconType.warning)({size: 100, color: colors.warning})}
        </View>
      ),
    }[type]);

  return (
    <View style={styles.container}>
      <View />
      <View style={styles.infoWrapper}>
        {renderIconFeedback(type)}
        <Text type={TextType.title} style={styles.text}>
          {title}
        </Text>
        <Text type={TextType.text} style={styles.text}>
          {description}
        </Text>
      </View>
      <Button type={ButtonType.primary} text={buttonText} onPress={onPress} />
    </View>
  );
};

export default Feedback;
