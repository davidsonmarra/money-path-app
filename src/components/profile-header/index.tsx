import React from 'react';
import {TouchableOpacity, View} from 'react-native';
import {SafeAreaView, SafeAreaViewProps} from 'react-native-safe-area-context';
import {Text, TextType} from 'src/components';
import useStyles from 'src/components/profile-header/styles';

interface Props extends SafeAreaViewProps {
  name: string;
  abbreviation: string;
  onPressProfileImage: () => void;
}

const ProfileHeader = ({name, abbreviation, onPressProfileImage}: Props) => {
  const style = useStyles();

  return (
    <SafeAreaView style={style.container} edges={['top']}>
      <View style={style.infoContainer}>
        <Text type={TextType.text} style={style.textColor}>
          Ol√°,{' '}
        </Text>
        <Text type={TextType.textBold} style={style.textColor}>
          {name}
        </Text>
      </View>
      <TouchableOpacity
        onPress={onPressProfileImage}
        style={useStyles().profileImage}
        testID="profile-image">
        <Text type={TextType.textBold} style={useStyles().textNameAbbreviation}>
          {abbreviation}
        </Text>
      </TouchableOpacity>
    </SafeAreaView>
  );
};

export default ProfileHeader;
