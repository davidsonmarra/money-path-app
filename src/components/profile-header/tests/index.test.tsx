import React from 'react';
import {render} from 'src/configs/test-utils';
import {ProfileHeader} from 'src/components';

const mockOnPressProfileImage = jest.fn();
const renderComponent = ({
  name = '',
  abbreviation = '',
  onPressProfileImage = mockOnPressProfileImage,
}) =>
  render(
    <ProfileHeader
      name={name}
      abbreviation={abbreviation}
      onPressProfileImage={mockOnPressProfileImage}
    />,
  );

describe('ProfileHeader', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('renders correctly', () => {
    const tree = renderComponent({}).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders name correctly', () => {
    const name = 'John Doe';
    const profileHeader = renderComponent({name});
    expect(profileHeader.findByText(name)).toBeTruthy();
  });

  it('renders abbreviation correctly', () => {
    const abbreviation = 'JD';
    const profileHeader = renderComponent({abbreviation});
    expect(profileHeader.findByText(abbreviation)).toBeTruthy();
  });

  it('calls onPressProfileImage correctly', () => {
    const profileHeader = renderComponent({});
    profileHeader.getByTestId('profile-image').props.onClick();
    expect(mockOnPressProfileImage).toHaveBeenCalledTimes(1);
  });
});
