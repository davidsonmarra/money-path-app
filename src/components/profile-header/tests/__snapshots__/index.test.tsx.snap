// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileHeader renders correctly 1`] = `
<RNCSafeAreaView
  edges={
    {
      "bottom": "additive",
      "left": "additive",
      "right": "additive",
      "top": "additive",
    }
  }
>
  <RNCSafeAreaView
    edges={
      {
        "bottom": "off",
        "left": "off",
        "right": "off",
        "top": "additive",
      }
    }
    style={
      {
        "backgroundColor": "#2A5C99",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingHorizontal": 24,
        "paddingVertical": 16,
        "width": "100%",
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <Text
        style={
          [
            {
              "color": "#D9D9D9",
              "fontSize": 16,
              "fontWeight": "normal",
            },
            {
              "color": undefined,
            },
          ]
        }
      >
        Ol√°,
         
      </Text>
      <Text
        style={
          [
            {
              "color": "#D9D9D9",
              "fontSize": 16,
              "fontWeight": "bold",
            },
            {
              "color": undefined,
            },
          ]
        }
      />
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#EAC43D",
          "borderRadius": 24,
          "height": 48,
          "justifyContent": "center",
          "opacity": 1,
          "width": 48,
        }
      }
      testID="profile-image"
    >
      <Text
        style={
          [
            {
              "color": "#D9D9D9",
              "fontSize": 16,
              "fontWeight": "bold",
            },
            {
              "color": "#2A5C99",
            },
          ]
        }
      />
    </View>
  </RNCSafeAreaView>
</RNCSafeAreaView>
`;
